debugstatus/handler.go

```diff
@@ -0,0 +1,96 @@
+// Copyright 2015 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+package debugstatus
+
+import (
+	"net/http"
+
+	pprof "github.com/juju/httpprof"
+	"github.com/juju/httprequest"
+)
+
+// Version describes the current version of the code being run.
+type Version struct {
+	GitCommit string
+	Version   string
+}
+
+// Handler implements a type that can be used with httprequest.Handlers
+// to serve a standard set of /debug endpoints, including
+// the version of the system, its current health status
+// the runtime profiling information.
+type Handler struct {
+	// Check will be called to obtain the current health of the
+	// system. It should return a map as returned from the
+	// Check function.
+	Check func() map[string]CheckResult
```

Comment:
I think the Handler shouldn't require you to set all of these things to use it. If Check of CheckPprofAllowed is nil then we should do something that isn't panicking. /debug/check could just return an empty set (not sure if that should be considered successful or not) CheckPprofAllowed could just deny access.